test/spec/modules/prebidServerBidAdapter_spec.js

```diff
@@ -506,6 +506,19 @@ describe('S2S Adapter', function () {
       resetSyncedStatus();
     });
 
+    it('should block request if config did not define proper URL in endpoint object config', function() {
+      let badConfig = utils.deepClone(CONFIG);
+      badConfig.endpoint = { noP1Consent: 'https://prebid.adnxs.com/pbs/v1/openrtb2/auction' };
+      config.setConfig({ s2sConfig: badConfig });
+
+      let badCfgRequest = utils.deepClone(REQUEST);
+      badCfgRequest.s2sConfig = badConfig;
+
+      adapter.callBids(badCfgRequest, BID_REQUESTS, addBidResponse, done, ajax);
+
+      expect(server.requests.length).to.equal(0);
+    });
+
```

Comment:
Maybe add two more test cases

1. for `p1Consent` only configuration (which is probably the more common one)
2. for an empty object


```suggestion
    [
      { p1Consent: 'https://prebid.adnxs.com/pbs/v1/openrtb2/auction' },
      { noP1Consent: 'https://prebid.adnxs.com/pbs/v1/openrtb2/auction' },
      { }
    ].forEach(endpoint => {
        it('should block request if config did not define proper URL in endpoint object config', function() {
          let badConfig = utils.deepClone(CONFIG);
          badConfig.endpoint = endpoint;
          config.setConfig({ s2sConfig: badConfig });

          let badCfgRequest = utils.deepClone(REQUEST);
          badCfgRequest.s2sConfig = badConfig;

          adapter.callBids(badCfgRequest, BID_REQUESTS, addBidResponse, done, ajax);

          expect(server.requests.length).to.equal(0);
        });
    });
```

Reply:
Do you mean add a test for when p1Consent is the only property defined and the request contains a no-consent scenario - to test that the request is blocked here as well?

Reply:
yes :)

Reply:
Ok thanks for confirming.  I added some tests guessing that was the case.  Please take a look when you have the chance (in addition to the other change you suggested).